qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/call_G2sc.py

```diff
@@ -0,0 +1,171 @@
+                                                        ))
+
+dmin = 1.0
+peaks_to_add = set()
+if refinement_method == "Pawley" and mantid_pawley_reflections:
+    for gsas_phase in gsas_phases:
+        gsas_phase.data['General']['doPawley'] = True
+
+        gsas_reflections = []
+        for reflection in mantid_pawley_reflections:
+            [h, k, l], d, multiplicity = reflection
+            gsas_reflections.append([int(h), int(k), int(l), int(multiplicity), float(d), True, 100.0, 1.0])
+        gsas_phase.data["Pawley ref"] = gsas_reflections
+
+
+# for i in G2sc.dictDive(phase.data['General'], 'paw'): print(i)
+
+# increase # of cycles to improve convergence
+gpx.data['Controls']['data']['max cyc'] = 3  # not in API
+
+# tutorial step 4: turn on background refinement (Hist)
+for index, histogram in enumerate(gsas_histograms):
+    if refine_background:
+        histogram.set_refinements({"Background": {"no. coeffs": 3, "refine": True}})
+    else:
+        histogram.set_refinements({"Background": {"no. coeffs": 0, "refine": False}})
+
+if not refine_histogram_scale_factor:
+    for gsas_histogram in gpx.histograms():
+        gsas_histogram.SampleParameters["Scale"] = [1.0, False]
+
+for gsas_phase in gpx.phases():
+    gsas_phase.set_refinements({"Cell": True})
+    if override_cell_lengths:
+        gsas_phase.data['General']['Cell'][1:4] = tuple(override_cell_lengths)
+
+if x_min and x_max:
+    for index, histogram in enumerate(gsas_histograms):
+        histogram.set_refinements({'Limits': [x_min[index], x_max[index]]})
+
+gpx.save(project_path)
+gpx.do_refinements()
+gpx.save(project_path)
+HistStats(gpx)
+
+if refine_microstrain:
+    for gsas_phase in gpx.phases():
+        gsas_phase.set_HAP_refinements({'Mustrain': { 'type': 'isotropic', 'refine': True}})
+    print("Refining Microstrain")
+    gpx.do_refinements()
+    gpx.save(project_path)
+    HistStats(gpx)
+
+if refine_sigma_one:
+    for gsas_histogram in gpx.histograms():
+        gsas_histogram.set_refinements({'Instrument Parameters': ['sig-1']})
+    print("Refining Sigma-1")
+    gpx.do_refinements()
+    gpx.save(project_path)
+    HistStats(gpx)
+
+if refine_gamma:
+    for gsas_histogram in gpx.histograms():
+        gsas_histogram.set_refinements({'Instrument Parameters': ['Y']})
+    print("Refining Gamma")
+    gpx.do_refinements()
+    gpx.save(project_path)
+    HistStats(gpx)
+
+for index, histogram in enumerate(gpx.histograms()):
+    histogram.Export(os.path.join(temporary_save_directory, project_name + f"_{index}.csv"), ".csv", "histogram CSV file")
+
+    # Assuming only one phase
+if refinement_method == 'Pawley':
+    phase_name = list(histogram.reflections().keys())[0]
+    reflection_positions = np.transpose(np.array(histogram.reflections()[phase_name]['RefList']))[5]
+    np.savetxt(os.path.join(temporary_save_directory, project_name + f"_reflections_{index}.txt"), reflection_positions)
```

Comment:
Do you need to import numpy for this?? One fewer dependency would be nice! (See FileIO Pluralsight course)